projects/engine/src/include/Core/Object.h

```diff
@@ -4,21 +4,25 @@
 
 namespace MAD
 {
-	using ObjectID = uint64_t;
+	class OGameWorld;
 
+	using ObjectID = uint64_t;
+	
 	class UObject
 	{
 		MAD_DECLARE_BASE_CLASS(UObject)
-
 	public:
-		UObject();
+		explicit UObject(OGameWorld* inOwningGameWorld = nullptr);
```

Comment:
This kind of comes down to preference, but I feel like we don't want to give `inOwningGameWorld` a default value. We kind of want it to be explicit that the caller knows they must provide a game world or purposefully pass `nullptr` right?